### PR TITLE
implement getRegistrationData API

### PR DESCRIPTION
**Pull request by mudulo**

### CODE REVIEW

File: CompassRNSampleApp/yarn.lock
Comment (muruthi) [inline CompassRNSampleApp/yarn.lock:1]: Was this change required?

File: android/src/main/java/com/mastercard/compass/cp3/lib/react_native_wrapper/route/GetRegistrationDataAPIRoute.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+class GetRegistrationDataAPIRoute(private val context: ReactApplicationContext, private val currentActivity: Activity?) {
+  companion object {
+    val REQUEST_CODE_RANGE = 700 until 800
+
+    const val GET_REGISTRATION_DATA_REQUEST_CODE = 700
+    private const val TAG = "GetRegistrationDataAPIRoute"
+  }
+
+  fun startGetRegistrationIntent(GetRegistrationDataParams: ReadableMap){
```
Comment (muruthi) [inline android/src/main/java/com/mastercard/compass/cp3/lib/react_native_wrapper/route/GetRegistrationDataAPIRoute.kt:25]: Fix Code Smells flagged by Sonar Cloud

**Review by muruthi — CHANGES_REQUESTED** [review]

File: CompassRNSampleApp/yarn.lock
Comment (mudulo) [reply CompassRNSampleApp/yarn.lock:1]: I think this file was auto generated when I run the yarn command, following the contributing.md steps, but I can add it to .gitignore for now

Comment (mudulo) [reply CompassRNSampleApp/yarn.lock:1]: but i see we are advised to add it to git [https://github.com/yarnpkg/yarn/issues/1583](https://github.com/yarnpkg/yarn/issues/1583), @Harry3167 can advise

Comment (muruthi) [reply CompassRNSampleApp/yarn.lock:1]: Confirm with @Harry3167 if thats the appropriate approach.

File: android/src/main/java/com/mastercard/compass/cp3/lib/react_native_wrapper/route/GetRegistrationDataAPIRoute.kt (hunk 1 above)
Comment (mudulo) [reply android/src/main/java/com/mastercard/compass/cp3/lib/react_native_wrapper/route/GetRegistrationDataAPIRoute.kt:25]: okay, sure

Comment (mudulo) [reply android/src/main/java/com/mastercard/compass/cp3/lib/react_native_wrapper/route/GetRegistrationDataAPIRoute.kt]: Fixed [Here](https://github.com/Mastercard/community-pass-react-native-wrapper/pull/5/commits/4bd23f105ef7819841ec3c04026c7b3edb466867)

File: CompassRNSampleApp/yarn.lock
Comment (Harry3167) [reply CompassRNSampleApp/yarn.lock:1]: The file is required, it contains a list of all dependencies including nested deps from other libraries.

File: docs/api-references/device-lifecycle-management-api.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+**Type Aliases**
+
+```ts
+// SaveBiometricConsentParamType
+interface GetRegistrationDataParamType {
+  reliantGUID: string;
+  programGUID: string;
+}
+
+// SaveBiometricConsentResultType
```
Comment (Harry3167) [inline docs/api-references/device-lifecycle-management-api.md:38]: Please update this to: // GetRegistrationDataResultType

**Review by muruthi — APPROVED** [review]

File: docs/api-references/device-lifecycle-management-api.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+**Response Parameters**
+| **Parameter** | **Type** | **Description** |
+|-----------------|-----------------|----------------------------------------------------------|
+| getRegistrationDataResponse | Promise<`GetRegistrationDataResultType`> | A promise that resolves to an object containing either an authMethods array with a boolean field indicating whether the user is registered in the current program, or an error field. |
+
+**Type Aliases**
+
+```ts
+// SaveBiometricConsentParamType
```
Comment (Harry3167) [inline docs/api-references/device-lifecycle-management-api.md:32]: Please update this to: // GetRegistrationDataParamType

File: android/src/main/java/com/mastercard/compass/cp3/lib/react_native_wrapper/route/GetRegistrationDataAPIRoute.kt [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    val reliantAppGUID: String = GetRegistrationDataParams.getString("reliantGUID")!!
```
Comment (Harry3167) [inline android/src/main/java/com/mastercard/compass/cp3/lib/react_native_wrapper/route/GetRegistrationDataAPIRoute.kt]: Please rename reliantAppGUID to reliantGUID for naming consistency across the files

**Review by Harry3167 — CHANGES_REQUESTED** [review]

**Comment by Harry3167** [comment]
Hey @mudulo I have suggested minor improvements, please have a look at them

File: docs/api-references/device-lifecycle-management-api.md (hunk 3 above)
Comment (mudulo) [reply docs/api-references/device-lifecycle-management-api.md]: Thanks, Corrected.

File: docs/api-references/device-lifecycle-management-api.md (hunk 2 above)
Comment (mudulo) [reply docs/api-references/device-lifecycle-management-api.md]: Thanks, Corrected
